From 580ad4fc6269a8ab85e9136a00d3738037ec0303 Mon Sep 17 00:00:00 2001
From: Daniel Nicoletti <dantti12@gmail.com>
Date: Fri, 30 Nov 2012 19:25:15 -0200
Subject: [PATCH] aptcc: Fix trusted packages not being emitted when an untrusted package was available
 Backported to PackageKit 0.7.x series by Matthias Klumpp <matthias@tenstral.net>

---
 backends/aptcc/apt-intf.cpp |   27 +++++++++++++++++----------
 1 files changed, 17 insertions(+), 10 deletions(-)

--- a/backends/aptcc/apt-intf.cpp
+++ b/backends/aptcc/apt-intf.cpp
@@ -1430,20 +1430,29 @@
     if (untrusted.empty()) {
         return true;
     } else if (simulating) {
+        // We are just simulating and have untrusted packages emit them
+        // and return true to continue processing
         emitPackages(untrusted, PK_FILTER_ENUM_NONE, PK_INFO_ENUM_UNTRUSTED);
     }
 
-    if (pk_backend_get_bool(m_backend, "only_trusted") == false) {
-        g_debug ("Authentication warning overridden.\n");
-        return true;
+    if (pk_backend_get_bool(m_backend, "only_trusted")) {
+         // We are NOT simulating and have untrusted packages
+         // fail the transaction.
+         string warning("The following packages cannot be authenticated:\n");
+         warning += UntrustedList;
+         pk_backend_error_code(m_backend,
+                                   PK_ERROR_ENUM_CANNOT_INSTALL_REPO_UNSIGNED,
+                                   warning.c_str());
+         _error->Discard();
+
+         return false;
+    } else {
+         // We are NOT simulating and have untrusted packages
+         // But the user didn't set ONLY_TRUSTED flag
+         g_debug ("Authentication warning overridden.\n");
+         return true;
     }
 
-    string warning("The following packages cannot be authenticated:\n");
-    warning += UntrustedList;
-    pk_backend_error_code(m_backend,
-                          PK_ERROR_ENUM_CANNOT_INSTALL_REPO_UNSIGNED,
-                          warning.c_str());
-    _error->Discard();
     return false;
 }
 
